ray-curator/ray_curator/stages/deduplication/fuzzy/minhash.py

```diff
@@ -0,0 +1,326 @@
+        char_ngrams: int = 24,
+        num_hashes: int = 260,
+        seed: int = 42,
+        use_64bit_hash: bool = False,
+        read_format: Literal["jsonl", "parquet"] = "jsonl",
+        read_kwargs: dict[str, Any] | None = None,
+        write_kwargs: dict[str, Any] | None = None,
+        pool: bool = True,
+    ):
+        # Initialize parent classes
+        ProcessingStage.__init__(self)
+        DeduplicationIO.__init__(self, id_generator=None)
+
+        # Set ProcessingStage attributes
+        self._name = self.__class__.__name__
+        self._resources = Resources(gpus=1.0)  # Requires 1 GPU
+
+        self.output_dir = output_dir
+        self.text_column = text_column
+        self.minhash_column = minhash_column
+        self.char_ngrams = char_ngrams
+        self.num_hashes = num_hashes
+        self.seed = seed
+        self.use_64bit_hash = use_64bit_hash
+        self.read_format = read_format
+        self.read_kwargs = read_kwargs or {}
+        self.write_kwargs = write_kwargs or {}
+        self.pool = pool
+        # Initialize the minhash processor in setup
+        self.minhash_processor = None
+        self.id_generator = None
+
+    def setup(self, _worker_metadata: "WorkerMetadata | None" = None) -> None:
+        """Initialize the GPU MinHash processor and ID generator."""
+        # Initialize the ID generator (will be shared across workers)
+        self.id_generator = IdGenerator.options(
+            name=CURATOR_ID_GENERATOR_ACTOR_NAME, get_if_exists=True, lifetime="detached"
+        ).remote()
+
+        # Initialize the GPU minhash processor
+        self.minhash_processor = GPUMinHash(
+            seed=self.seed,
+            num_hashes=self.num_hashes,
+            char_ngrams=self.char_ngrams,
+            use_64bit_hash=self.use_64bit_hash,
+            pool=self.pool,
+        )
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        """Define input requirements."""
+        return (["data"], [])
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        """Define outputs - produces FileGroupTask with minhash files."""
+        return (["data"], [])
+
+    def process(self, task: FileGroupTask) -> FileGroupTask:
+        """
+        Process a group of files to compute minhashes.
+
+        Args:
+            task: FileGroupTask containing file paths to process
+
+        Returns:
+            FileGroupTask containing paths to minhash output files
+        """
+
+        if self.minhash_processor is None or self.id_generator is None:
+            msg = "MinHash processor or ID generator not initialized. Call setup() first."
+            raise RuntimeError(msg)
+
+        fs = get_fs(self.output_dir, self.write_kwargs.get("storage_options", {}))
+        output_file = fs.sep.join([self.output_dir, self._name, f"{task.task_id}.parquet"])
+
+        read_kwargs = self.read_kwargs.copy()
+        read_kwargs["storage_options"] = task._metadata.get("storage_options", {})
```

Comment:
I'm sus about passing in storage_options from `task_metadata`. I think it bloats, likely leaks keys etc. I know we have it there, but I'm curious why not lead any reader just accept storage_option as a key?

Reply:
I'm okay with that approach as well. The main motivation was this:
For general readers (like jsonl) which are compositeStages you provide the `storage_options` once and that'll be used for both FilePartitioningStage as well as JsonlReaderStage.

Since there's no composite here the user experience would be
```
FilePartitioningStage(storage_options)
MinHashStage(same_storage_options)
```
I'm fine with that if reusing from the prev task can lead to unexpected behavior.